### DISCUSSION TITLE
Why not use use-sync-external-store directly from react?

### DISCUSSION THREAD

**Discussion by Minozzzi** [pmndrs/zustand#2181] [General]
For example, in the line
https://github.com/pmndrs/zustand/blob/a836356b5e4062abde852af3ba5ee6fc3f1a41fe/src/react.ts#L18-L19

Can't it be done like this?
```ts
const { useDebugValue, useSyncExternalStore } = ReactExports 
const { useSyncExternalStoreWithSelector } = useSyncExternalStoreExports 
 ```

**Comment by dbritto-dev** [comment]
@Minozzzi `useSyncExternalStore` does not support selectors that's why we use `use-sync-external-store` package

**Reply by dai-shi** [reply]
selector is actually fine, but equalityFn is missing.

Also, we use `use-sync-external-store` to support React 17.

**Reply by dbritto-dev** [reply]
@dai-shi yeah, `equalityFn`
